cmd/fs-v1.go

```diff
@@ -217,7 +217,62 @@ func (fs fsObjects) DeleteBucket(bucket string) error {
 
 /// Object Operations
 
-// GetObject - get an object.
+// CopyObject - copy object source object to destination object.
+func (fs fsObjects) CopyObject(srcBucket, srcObject, dstBucket, dstObject string, metadata map[string]string, updateMetadata bool) (ObjectInfo, error) {
+	// Stat the file to get file size.
+	fi, err := fs.storage.StatFile(srcBucket, srcObject)
+	if err != nil {
+		return ObjectInfo{}, toObjectErr(traceError(err), srcBucket, srcObject)
+	}
+
```

Comment:
if `updateMetadata` is true but src and dst are different the dst object is not written, just the metadata gets written.